terminus-ui/chip/src/chip.component.ts

```diff
@@ -0,0 +1,432 @@
+      ? this.elementRef.nativeElement.textContent : this._value;
+  }
+  public set value(value: string) {
+    this._value = value;
+  }
+  protected _value = '';
+
+  /**
+   * Whether or not the chip is selectable. By default a chip is
+   * selectable, and it becomes non-selectable if its parent chip collection is
+   * not selectable.
+   */
+  @Input()
+  public get isSelectable(): boolean {
+    return this._selectable && this.chipCollectionSelectable;
+  }
+  public set isSelectable(value: boolean) {
+    this._selectable = coerceBooleanProperty(value);
+  }
+  protected _selectable = true;
+
+  /**
+   * Determines whether or not the chip displays the remove styling and emits (removed) events.
+   */
+  @Input()
+  public get isRemovable(): boolean {
+    return this._removable;
+  }
+  public set isRemovable(value: boolean) {
+    this._removable = coerceBooleanProperty(value);
+    this.changeDetectorRef.detectChanges();
+  }
+  protected _removable = true;
+
+  /**
+   * Emitted when the chip is selected or deselected.
+   */
+  @Output()
+  public readonly selectionChange = new EventEmitter<TsChipSelectionChange>();
+
+  /**
+   * Emitted when the chip is destroyed.
+   */
+  @Output()
+  public readonly destroyed = new EventEmitter<TsChipEvent>();
+
+  /**
+   * Emitted when a chip is to be removed.
+   */
+  @Output()
+  public readonly removed = new EventEmitter<TsChipEvent>();
+
+  constructor(public elementRef: ElementRef,
+              private ngZone: NgZone,
+              private changeDetectorRef: ChangeDetectorRef) {
+    super(elementRef);
+
+  }
+
+  public ngOnDestroy() {
+    this.destroyed.emit({chip: this});
+    // eslint-disable-next-line no-underscore-dangle
+  }
+
+  /**
+   * Selects the chip.
+   */
+  public select(): void {
+    if (!this.selected) {
+      this.selected = true;
+      this.dispatchSelectionChange();
+    }
+  }
+
+  /**
+   * Deselects the chip.
+   */
+  public deselect(): void {
+    if (this.selected) {
+      this.selected = false;
+      this.dispatchSelectionChange();
+    }
+  }
+
+  /**
+   * Toggles the current selected state of this chip.
+   */
+  public toggleSelected(): boolean {
+    this._selected = !this.selected;
+    this.dispatchSelectionChange();
+    return this.selected;
+  }
+
+  /**
+   * Allows for programmatic focusing of the chip.
+   */
+  public focus(): void {
+    if (!this.hasFocus) {
+      this.elementRef.nativeElement.focus();
+      this.onFocus.next({chip: this});
+    }
+    this.hasFocus = true;
+  }
+
+  /**
+   * Allows for programmatic removal of the chip. Called by the TsChipCollectionComponent when the DELETE or
+   * BACKSPACE keys are pressed.
+   * {@link TsChipCollectionComponent}
+   *
+   * Informs any listeners of the removal request. Does not remove the chip from the DOM.
+   */
+  public remove(): void {
+    if (this.isRemovable) {
+      this.removed.emit({chip: this});
+    }
+  }
+
+  /**
+   * Handles click events on the chip.
```

Comment:
Need line break between title and param so our docs generation correctly lays it out